src/utils/requests.js

```diff
@@ -0,0 +1,51 @@
+import { useCallback } from "react";
+
+const base_url="https://localhost:8000";
+
+const sendRequest = useCallback(
+  async (url, method = "GET", body = null, headers = {}) => {
+    try {
+      const response = await fetch(base_url+ url, {
+        method,
+        body,
+        headers,
+      });
+      const responseData = await response.json();
+      if (!response.ok) throw new Error(responseData.status);
+    } catch (err) {
+      console.log(err);
+    }
+    return { responseData };
```

Comment:
Throwing an error inside the try-catch block will just take you to the catch block. Instead, if you want to return the response status code in case of request failure, remove the try-catch.